keps/sig-network/2091-cluster-network-policy/README.md

```diff
@@ -0,0 +1,999 @@
+  determine whether a connection matching a particular rule/policy must be
+  logged or not.
+- **Rule identifier**: In order to collect traffic statistics corresponding to
+  a rule, it is necessary to identify the rule which allows/denies that traffic.
+  This helps administrators figure the impact of the rules written in a
+  cluster-scoped policy resource. Thus, the ability to uniquely identify a rule
+  within a cluster-scoped policy resource becomes very important.
+  This can be addressed by introducing a field, say `name`, per `ClusterNetworkPolicy`
+  and `ClusterDefaultNetworkPolicy` ingress/egress rule.
+- **Node Selector**: Cluster administrators and developers want to write
+  policies that apply to cluster nodes or host network pods. This can be
+  addressed by introducing nodeSelector field under `appliedTo` field of the
+  `ClusterNetworkPolicy` and `ClusterDefaultNetworkPolicy` spec.
+  `ClusterDefaultNetworkPolicy` is a better candidate compared to K8s `NetworkPolicy`
+  for introducing this field as nodes are cluster level resources.
+
+## Design Details
+
+### ClusterNetworkPolicy API Design
+The following new `ClusterNetworkPolicy` API will be added to the `networking.k8s.io` API group:
+
+```golang
+type ClusterNetworkPolicy struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	Spec ClusterNetworkPolicySpec
+}
+
+type ClusterNetworkPolicySpec struct {
+	// No implicit isolation of AppliedTo Pods.
+	AppliedTo    AppliedTo
+	Ingress      []ClusterNetworkPolicyIngressRule
+	Egress       []ClusterNetworkPolicyEgressRule
+}
+
+type ClusterNetworkPolicyIngress/EgressRule struct {
+	Action       RuleAction
+	Ports        []networkingv1.NetworkPolicyPort
+	From/To      []ClusterNetworkPolicyPeer
+}
+
+type ClusterNetworkPolicyPeer struct {
+	PodSelector  *metav1.LabelSelector
+	// required if a PodSelector is specified
+	Namespaces   *Namespaces
+	IPBlock      *networkingv1.IPBlock
+}
+
+const (
+  RuleActionAuthorize RuleAction = "Authorize"
+	RuleActionDeny      RuleAction = "Deny"
+	RuleActionAllow     RuleAction = "Allow"
+)
+```
+
+For the ClusterNetworkPolicy ingress/egress rule, the `Action` field dictates whether
+traffic should be allowed or denied or always allowed from/to the ClusterNetworkPolicyPeer.
+This will be a required field.
+
+### ClusterDefaultNetworkPolicy API Design
+The following new `ClusterDefaultNetworkPolicy` API will be added to the `networking.k8s.io` API group:
+
+```golang
+type ClusterDefaultNetworkPolicy struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	Spec ClusterDefaultNetworkPolicySpec
+}
+
+type ClusterDefaultNetworkPolicySpec struct {
+	// Implicit isolation of AppliedTo Pods.
+	AppliedTo   AppliedTo
+	Ingress     []ClusterDefaultNetworkPolicyIngressRule
+	Egress      []ClusterDefaultNetworkPolicyEgressRule
+}
+
+type ClusterDefaultNetworkPolicyIngress/EgressRule struct {
+	Ports            []networkingv1.NetworkPolicyPort
+	OnlyFrom/OnlyTo  []ClusterDefaultNetworkPolicyPeer
```

Comment:
> We chose this naming to better hint policy writers about the isolation effect of ClusterDefaultNetworkPolicy on the workloads it applies to.

To me, this feels like an unnecessary divergence from the existing NP API that will just trip up policy writers by forcing them to remember syntactical differences between ClusterDefaultNetworkPolicy and NetworkPolicy that are not consequential to the way the two resources behave.

Reply:
Okay I will change this for the sake of consistency

Reply:
removed text